src/main/java/redis/clients/jedis/ConnectionFactory.java

```diff
@@ -35,6 +35,11 @@ public ConnectionFactory(final JedisSocketFactory jedisSocketFactory, final Jedi
     this.jedisSocketFactory = jedisSocketFactory;
   }
 
+  /**
+   * @deprecated Use {@link RedisCredentialsProvider} through
+   * {@link JedisClientConfig#getCredentialsProvider()}.
+   */
+  @Deprecated
```

Comment:
Why do we have to deprecate this? IMHO we can still keep the interface.

Reply:
This method<sup>[1]</sup> has been a headache for a very long time. We attempted several times to remove it but couldn't only to support Sentinel mode. Now handling it through RedisCredentialsProvider is much more saner way.

Reply:
This method has been a headache for a very long time. We attempted several times to remove it but couldn't only to support Sentinel mode. Now handling it through RedisCredentialsProvider is much more saner way.